setup.py

```diff
@@ -11,9 +11,20 @@
 SRC_DIR = os.path.join(THIS_DIR, "src")
 SNOWPARK_SRC_DIR = os.path.join(SRC_DIR, "snowflake", "snowpark")
 CONNECTOR_DEPENDENCY_VERSION = ">=3.2.0, <4.0.0"
+INSTALL_REQ_LIST = [
+    "setuptools>=40.6.0",
+    "wheel",
+    "cloudpickle>=1.6.0,<=2.0.0",
+    f"snowflake-connector-python{CONNECTOR_DEPENDENCY_VERSION}",
+    "pyyaml",
+]
 REQUIRED_PYTHON_VERSION = ">=3.8, <3.11"
 if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):
     REQUIRED_PYTHON_VERSION = ">=3.8"
+    # replace the line with condition py < 3.11
+    INSTALL_REQ_LIST[2] = "cloudpickle>=1.6.0,<=2.0.0;python_version<'3.11'"
+    # add one more line for py ~= 3.11
+    INSTALL_REQ_LIST.append("cloudpickle>=2.2.1,<3.0.0;python_version~='3.11'")
```

Comment:
2.2.1 is available in the anaconda channel, but not 2.2.0, 2.1.0. cloudpickle starts supporting python 3.11 from 2.1.0.

checking with ananconda team to see if it's on purpose that 2.1.0/2.2.0 are skipped